PathPlanning/RRT/sobol/sobol.py

```diff
@@ -0,0 +1,911 @@
+
+        v[37:40, 7] = np.transpose([7, 23, 39])
+        #
+        #    Set POLY.
+        #
+        poly = [
+            1, 3, 7, 11, 13, 19, 25, 37, 59, 47, 61, 55, 41, 67, 97, 91, 109,
+            103, 115, 131, 193, 137, 145, 143, 241, 157, 185, 167, 229, 171,
+            213, 191, 253, 203, 211, 239, 247, 285, 369, 299
+        ]
+
+        atmost = 2**log_max - 1
+        #
+        #    Find the number of bits in ATMOST.
+        #
+        maxcol = i4_bit_hi1(atmost)
+        #
+        #    Initialize row 1 of V.
+        #
+        v[0, 0:maxcol] = 1
+
+        # Things to do only if the dimension changed.
+
+    if dim_num != dim_num_save:
+        #
+        #    Check parameters.
+        #
+        if (dim_num < 1 or dim_max < dim_num):
+            print('I4_SOBOL - Fatal error!')
+            print('    The spatial dimension DIM_NUM should satisfy:')
+            print('        1 <= DIM_NUM <= %d' % dim_max)
+            print('    But this input value is DIM_NUM = %d' % dim_num)
+            return None
+
+        dim_num_save = dim_num
+        #
+        #    Initialize the remaining rows of V.
+        #
+        for i in range(2, dim_num + 1):
+            #
+            #    The bits of the integer POLY(I) gives the form of polynomial
+            #    I.
+            #
+            #    Find the degree of polynomial I from binary encoding.
+            #
+            j = poly[i - 1]
+            m = 0
+            while True:
+                j = math.floor(j / 2.)
+                if (j <= 0):
+                    break
+                m = m + 1
+            #
+            #    Expand this bit pattern to separate components of the logical
+            #    array INCLUD.
+            #
+            j = poly[i - 1]
+            includ = np.zeros(m)
+            for k in range(m, 0, -1):
+                j2 = math.floor(j / 2.)
+                includ[k - 1] = (j != 2 * j2)
+                j = j2
+            #
+            #    Calculate the remaining elements of row I as explained
+            #    in Bratley and Fox, section 2.
+            #
+            for j in range(m + 1, maxcol + 1):
+                newv = v[i - 1, j - m - 1]
+                l_var = 1
+                for k in range(1, m + 1):
+                    l_var = 2 * l_var
+                    if (includ[k - 1]):
+                        newv = np.bitwise_xor(
+                            int(newv), int(l_var * v[i - 1, j - k - 1]))
+                v[i - 1, j - 1] = newv
+#
+#    Multiply columns of V by appropriate power of 2.
+#
+        l_var = 1
+        for j in range(maxcol - 1, 0, -1):
+            l_var = 2 * l_var
+            v[0:dim_num, j - 1] = v[0:dim_num, j - 1] * l_var
+#
+#    RECIPD is 1/(common denominator of the elements in V).
+#
+        recipd = 1.0 / (2 * l_var)
+        lastq = np.zeros(dim_num)
+
+    seed = int(math.floor(seed))
+
+    if (seed < 0):
+        seed = 0
+
+    if (seed == 0):
+        l_var = 1
```

Comment:
lgtm.com reports this line as unnecessary code, because `l_var` is already 1 in line 417.